src/Microsoft.Identity.Abstractions/DownstreamRestApi/AuthorizationHeaderProviderOptions.cs

```diff
@@ -0,0 +1,140 @@
+﻿// Licensed under the MIT License.
+
+using System;
+using System.ComponentModel;
+using System.Net.Http;
+
+namespace Microsoft.Identity.Abstractions
+{
+    /// <summary>
+    /// Options passed-in to call downstream web APIs. To call Microsoft Graph, see rather
+    /// <c>MicrosoftGraphOptions</c> in the <c>Microsoft.Identity.Web.MicrosoftGraph</c> assembly.
+    /// </summary>
+    public class AuthorizationHeaderProviderOptions
+    {
+        AcquireTokenOptions _acquireTokenOptions = new();
+        HttpMethod _httpMethod = HttpMethod.Get;
+        string _protocolScheme = "Bearer";
+
+        /// <summary>
+        /// Default constructor.
+        /// </summary>
+        public AuthorizationHeaderProviderOptions()
+        {
+        }
+
+        /// <summary>
+        /// Copy constructor for <see cref="DownstreamRestApiOptions"/>
+        /// </summary>
+        /// <param name="other">Options to copy from.</param>
+        /// <exception cref="ArgumentNullException">Thrown if <paramref name="other"/> is <c>null</c>.</exception>
+        public AuthorizationHeaderProviderOptions(AuthorizationHeaderProviderOptions other)
+        {
+            if (other == null)
+            {
+                throw new ArgumentNullException(nameof(other));
+            }
+
+            BaseUrl = other.BaseUrl;
+            RelativePath = other.RelativePath;
+            AcquireTokenOptions = other.AcquireTokenOptions.Clone();
+            HttpMethod = other.HttpMethod;
+            CustomizeHttpRequestMessage = other.CustomizeHttpRequestMessage;
+            ProtocolScheme = other.ProtocolScheme;
+            RequestAppToken = other.RequestAppToken;
+        }
+
+        /// <summary>
+        /// Base URL for the called downstream web API. For instance <c>"https://graph.microsoft.com/beta/"</c>.
+        /// </summary>
+        public string? BaseUrl { get; set; }
+
+        /// <summary>
+        /// Path relative to the <see cref="BaseUrl"/> (for instance "me").
+        /// </summary>
+        public string RelativePath { get; set; } = string.Empty;
+
+        /// <summary>
+        /// HTTP method used to call this downstream web API (by default Get).
+        /// </summary>
+        [DefaultValue("Get")]
+        public HttpMethod HttpMethod
+        {
+            get
+            {
+                return _httpMethod;
+            }
+            set
+            {
+                _httpMethod = value ?? throw new ArgumentNullException(nameof(value));
+            }
+        }
+
+        /// <summary>
+        /// Provides an opportunity for the caller app to customize the HttpRequestMessage. For example,
+        /// to customize the headers. This is called after the message was formed, including
+        /// the Authorization header, and just before the message is sent.
+        /// </summary>
+        public Action<HttpRequestMessage>? CustomizeHttpRequestMessage { get; set; }
```

Comment:
Suggest verb:noun for naming actions.
HttpRequestMessageCustomizer

Reply:
Done!